acceptancetests/assess_primary_sub_relations.py

```diff
@@ -0,0 +1,225 @@
+      - Remove relation from single primary app A
+      - Ensure unit is removed from A
+      - Ensure unit is not removed from B and remains operational
+    """
+    test_model = create_model_with_primary_and_sub_apps(
+        client, 'remove-subordinate')
+
+    ensure_subordinate_unit_operational(
+        test_model, primary_app_name='dummy-source')
+    ensure_subordinate_unit_operational(
+        test_model, primary_app_name='dummy-sink')
+
+    test_model.juju('remove-relation', ('ntp', 'dummy-source'))
+
+    assert_subordinate_is_removed(test_model, 'dummy-source', 'ntp')
+
+    # This subordinate must still remain operational, we didn't remove the
+    # relation from dummy-sink.
+    ensure_subordinate_unit_operational(
+        test_model, primary_app_name='dummy-sink')
+
+    # We expect this check to fail now as the subordinate app shouldn't be
+    # running.
+    # If it doesn't fail then the unit is still operational and that is a fail
+    # state.
+    try:
+        ensure_subordinate_unit_operational(
+            test_model, primary_app_name='dummy-source')
+    except JujuAssertionError:
+        log.info('Subordinate application relation successfully removed.')
+    else:
+        raise JujuAssertionError(
+            'Subordinate unit still operational after relation removal')
+
+    test_model.destroy_model()
+
+
+def assert_service_is_removed(client, application_name, timeout=60):
+    """Wait for the named application to be removed from status.
+
+    Check status of `client` until the named application `application_name` no
+    longer appears.
+    Will raise if the application still appears after the timeout.abs
+
+    :param client: ModelClient with which to check for application status.
+    :param application_name: String name of the application to assert for in
+      status.
+    :param timeout: int seconds to wait for raising exception if application
+      still exists.
+    :raises JujuAssertionError: When application_name still appears in status
+      after the timeout is encountered.
+    """
+
+    for _ in until_timeout(timeout):
+        current_status = client.get_status()
+        current_status.get_applications().keys()
+        if application_name not in current_status.get_applications().keys():
+            return
+        else:
+            time.sleep(1)
+    raise JujuAssertionError(
+        'Application "{}" failed to be removed in {} seconds'.format(
+            application_name, timeout))
+
+
+def assert_subordinate_is_removed(
+        client, primary_name, subordinate_name, timeout=60):
+    for _ in until_timeout(timeout):
+        status = client.get_status()
+        subordinate_names = [
+            unit[0] for unit in status.service_subordinate_units(
+                primary_name)
+            ]
+        if not any(subordinate_name in unit for unit in subordinate_names):
+            return
+        else:
+            time.sleep(1)
```

Comment:
This might be a bit quick eh? Sleeping for 5 seconds is probably still fine. Probably not an issue, but I think we generally sleep longer than this? Don't feel the need to change, just asking.